pkg/catalog/routes.go

```diff
@@ -679,34 +682,51 @@ func (mc *MeshCatalog) buildOutboundPolicies(source service.K8sServiceAccount, t
 	return outPolicies
 }
 
-// listPoliciesFromTrafficTargets loops through all SMI Traffic Target resources and returns inbound and outbound traffic policies
-// based on when the given service account matches a destination or source in the Traffic Target resource
-func (mc *MeshCatalog) listPoliciesFromTrafficTargets(sa service.K8sServiceAccount) ([]*trafficpolicy.InboundTrafficPolicy, []*trafficpolicy.OutboundTrafficPolicy, error) {
+// listInboundPoliciesFromTrafficTargets builds inbound traffic policies for all inbound services
+// when the given service account matches a destination in the Traffic Target resource
+func (mc *MeshCatalog) listInboundPoliciesFromTrafficTargets(upstreamIdentity service.K8sServiceAccount, upstreamServices []service.MeshService) []*trafficpolicy.InboundTrafficPolicy {
 	inboundPolicies := []*trafficpolicy.InboundTrafficPolicy{}
-	outboundPolicies := []*trafficpolicy.OutboundTrafficPolicy{}
 
 	for _, t := range mc.meshSpec.ListTrafficTargets() { // loop through all traffic targets
 		if !isValidTrafficTarget(t) {
 			continue
 		}
 
-		if t.Spec.Destination.Name == sa.Name { // found inbound
-			inboundPolicies = trafficpolicy.MergeInboundPolicies(false, inboundPolicies, mc.buildInboundPolicies(t)...)
+		if t.Spec.Destination.Name != upstreamIdentity.Name { // not an inbound policy for service
```

Comment:
Where are we now checking the the name in the destination is indeed the upstream identity since the if condition has been deleted?

Reply:
nvm brain fart